### PR TITLE
fix: crash when parsing invalid json config

### LINKED ISSUE #51: Crash in eppoclient::parseConfigResponse
Hello
We're seeing crash reports with these stacks, no repro...

```
EXCEPTION_ACCESS_VIOLATION_READ / 0x0: Fatal Error: EXCEPTION_ACCESS_VIOLATION_READ / 0x0
  chrome              0x7ff8c6a1376f std::__Cr::vector<T>::size (vector.h:398)
  chrome              0x7ff8c6a1376f std::__Cr::vector<T>::vector (vector.h:268)
  chrome              0x7ff8c6a1376f std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a1376f std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a1376f std::__Cr::allocator_traits<T>::construct (allocator_traits.h:302)
  chrome              0x7ff8c6a1376f nlohmann::json_abi_v3_11_3::basic_json<T>::create (json.hpp:19690)
  chrome              0x7ff8c6a1376f nlohmann::json_abi_v3_11_3::basic_json<T>::json_value::json_value (json.hpp:19839)
  chrome              0x7ff8c6a1376f nlohmann::json_abi_v3_11_3::basic_json<T>::basic_json (json.hpp:20464)
  chrome              0x7ff8c6a1447b eppoclient::Condition::Condition (config_response.hpp:70)
  chrome              0x7ff8c6a1447b std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a1447b std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a1447b std::__Cr::allocator_traits<T>::construct (allocator_traits.h:302)
  chrome              0x7ff8c6a1447b std::__Cr::__uninitialized_allocator_copy_impl (uninitialized_algorithms.h:473)
  chrome              0x7ff8c6a1447b std::__Cr::__uninitialized_allocator_copy (uninitialized_algorithms.h:512)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::__construct_at_end (vector.h:971)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::__init_with_size (vector.h:604)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::vector (vector.h:268)
  chrome              0x7ff8c6a1447b eppoclient::Rule::Rule (config_response.hpp:91)
  chrome              0x7ff8c6a1447b std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a1447b std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a1447b std::__Cr::allocator_traits<T>::construct (allocator_traits.h:302)
  chrome              0x7ff8c6a1447b std::__Cr::__uninitialized_allocator_copy_impl (uninitialized_algorithms.h:473)
  chrome              0x7ff8c6a1447b std::__Cr::__uninitialized_allocator_copy (uninitialized_algorithms.h:512)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::__construct_at_end (vector.h:971)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::__init_with_size (vector.h:604)
  chrome              0x7ff8c6a1447b std::__Cr::vector<T>::vector (vector.h:268)
  chrome              0x7ff8c6a1447b eppoclient::Allocation::Allocation (config_response.hpp:99)
  chrome              0x7ff8c6a1447b std::__Cr::construct_at<T> (construct_at.h:38)
  chrome              0x7ff8c6a132c7 std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a132c7 std::__Cr::allocator_traits<T>::construct (allocator_traits.h:302)
  chrome              0x7ff8c6a132c7 std::__Cr::__uninitialized_allocator_copy_impl (uninitialized_algorithms.h:473)
  chrome              0x7ff8c6a132c7 std::__Cr::__uninitialized_allocator_copy (uninitialized_algorithms.h:512)
  chrome              0x7ff8c6a132c7 std::__Cr::vector<T>::__construct_at_end (vector.h:971)
  chrome              0x7ff8c6a132c7 std::__Cr::vector<T>::__init_with_size (vector.h:604)
  chrome              0x7ff8c6a132c7 std::__Cr::vector<T>::vector (vector.h:268)
  chrome              0x7ff8c6a132c7 eppoclient::FlagConfiguration::FlagConfiguration (config_response.hpp:130)
  chrome              0x7ff8c6a132c7 std::__Cr::pair<T>::pair (pair.h:107)
  chrome              0x7ff8c6a1315f std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a1315f std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a1315f std::__Cr::allocator_traits<T>::construct (allocator_traits.h:302)
  chrome              0x7ff8c6a1315f std::__Cr::__hash_node<T>::__hash_node (__hash_table:173)
  chrome              0x7ff8c6a1315f std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a1315f std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a1315f std::__Cr::__hash_table<T>::__construct_node_hash (__hash_table:1857)
  chrome              0x7ff8c6a1315f std::__Cr::__hash_table<T>::__copy_construct (__hash_table:696)
  chrome              0x7ff8c6a1315f std::__Cr::__hash_table<T>::__copy_construct (__hash_table:720)
  chrome              0x7ff8c6a3a198 std::__Cr::__hash_table<T>::__hash_table (__hash_table:1131)
  chrome              0x7ff8c6a3a198 std::__Cr::unordered_map<T>::unordered_map (unordered_map:978)
  chrome              0x7ff8c6a3a198 eppoclient::ConfigResponse::ConfigResponse (config_response.hpp:151)
  chrome              0x7ff8c6a3a198 std::__Cr::construct_at (construct_at.h:38)
  chrome              0x7ff8c6a3a198 std::__Cr::__construct_at (construct_at.h:46)
  chrome              0x7ff8c6a3a198 std::__Cr::__optional_storage_base<T>::__construct (optional:391)
  chrome              0x7ff8c6a3a198 std::__Cr::optional<T>::operator= (optional:785)
  chrome              0x7ff8c6a3a198 eppoclient::parseConfigResponse (config_response.cpp:687)
  chrome              0x7ff8c6a3c17a eppoclient::parseConfigResponse (config_response.cpp:716)
  chrome              0x7ff8c6a1143f pplx::internal::EppoSdkWrapper::UpdateConfiguration (eppo_sdk_wrapper.cc:150)
 ...
```

Our code (bottom of the stack) is super simple:
```
bool EppoSdkWrapper::UpdateConfiguration(
    const std::string& flags_config_json,
    const std::string& bandits_config_json) {
  auto config_response = eppoclient::parseConfigResponse(flags_config_json);  // <-- crashes here
  if (!config_response.hasValue()) {
    return false;
  }
  ...
```

Value of `flags_config_json` during the crash is unknown, it's supposed to be the content downloaded from eppo backend, but it might be empty or corrupt somehow.

### PR DESCRIPTION
**Pull request by greghuels**
_Eppo Internal:_

[//]: #  (Link to the issue corresponding to this chunk of work)
:tickets: Fixes [FFESUPPORT-458](https://datadoghq.atlassian.net/browse/FFESUPPORT-458)

## Summary

Fixes #51 - crash in `parseConfigResponse` when receiving invalid or non-object JSON input.

- Added validation to check that input JSON is an object before calling `contains()` method
- Added comprehensive test coverage for malformed input edge cases

## Root Cause

The `parseConfigResponse(const nlohmann::json& j)` function called `j.contains("flags")` without first validating that `j` is a JSON object. In nlohmann::json, calling `contains()` on a non-object type (null, array, number, string, boolean) throws a `type_error.306` exception, causing the `EXCEPTION_ACCESS_VIOLATION_READ` crash reported in the issue.

This could occur when the backend returns empty, corrupt, or unexpected responses.

## Test plan

- [x] Added 11 new test cases covering edge cases:
  - Empty string input
  - Null JSON input (both as `json` object and string `"null"`)
  - Array JSON input (both as `json` object and string `"[]"`)
  - Number JSON input
  - String JSON input
  - Boolean JSON input
  - Malformed JSON string
  - Truncated JSON string
- [x] All 125 existing tests continue to pass

### CODE REVIEW

File: src/config_response.cpp [hunk 1]
```diff
@@ -627,6 +627,13 @@ ParseResult<ConfigResponse> parseConfigResponse(const nlohmann::json& j) {
     ParseResult<ConfigResponse> result;
     ConfigResponse cr;
 
+    // Validate that input is a JSON object
+    if (!j.is_object()) {
+        result.errors.push_back("ConfigResponse: Expected a JSON object, got " +
+                                std::string(j.type_name()));
```
Comment (leoromanovsky) [inline src/config_response.cpp:633]: consider another log line that prints out the received configuration to give them a totally self service way to iterate; alternatively does this error give them a strong enough signal that they could catch and print it themselves?

**Review by leoromanovsky — APPROVED** [review]

File: src/config_response.cpp (hunk 1 above)
Comment (greghuels) [reply src/config_response.cpp:633]: The current error message includes the JSON type name (e.g., "Expected a JSON object, got null" or "got array"), which should give a good signal about what happened. Since parseConfigResponse returns a ParseResult<ConfigResponse> with the error in result.errors, consuming apps can catch and log the full input themselves. So we should be good 👍
